### PR TITLE
[DX] Make tracy debug functions part of dev-only, so developers can use own debug style

### PR DESCRIPTION
**Pull request by TomasVotruba**
These functions should never be used as public API, as Tracy should not be enforced.

Each dev should is able to use their own functions and ways they're used to - Symfony debug, xdebug, Ray etc. :+1:

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -88,7 +88,6 @@
             "Rector\\Utils\\": "utils"
         },
         "files": [
-            "src/functions/node_helper.php",
```
Comment (samsonasik) [inline composer.json:91]: `src/functions` can be removed from downgrade workflow

https://github.com/rectorphp/rector-src/blob/4765d5a18552704431a5064a308b73a2414c2190/.github/workflows/build_scoped_rector.yaml#L55

Comment (TomasVotruba) [reply composer.json:91]: Done in main :+1:  Thanks :slightly_smiling_face:

Comment (samsonasik) [reply composer.json:91]: `vendor/symfony/string/Resources/functions.php` is needed to executed early

**Comment by TomasVotruba** [comment]
Side effect of this cleanup is 7000 lines less in the shipped in every Rector install :partying_face: 

image

**Comment by samsonasik** [comment]
@TomasVotruba I think in our `rector-*` extensions, we need to have debugger tool, if it goes to require-dev, then they need to have it in the require-dev as wel?

**Comment by TomasVotruba** [comment]
Yes, you can the tracy there if/when you need it.

**Comment by samsonasik** [comment]
To avoid repetitive create a `debug_functions.php`, I think we need to have a new repository intead, eg: `rector-debugging` which contains `debug_functions.php` content and if there is addition, we can add there instead of update to all repository, then finally, we can `require --dev rector/rector-debugging` in every repos, wdyt?

**Comment by TomasVotruba** [comment]
That could a way to go :+1:

**Comment by samsonasik** [comment]
Ok, I create a repository for it https://github.com/rectorphp/rector-debugging

**Comment by samsonasik** [comment]
I create init repo PR for it https://github.com/rectorphp/rector-debugging/pull/1

**Pull request merged**